core_impl/src/main/java/com/google/instrumentation/trace/SpanFactoryImpl.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.instrumentation.common.Clock;
+import com.google.instrumentation.trace.Link.Type;
+import com.google.instrumentation.trace.Span.Options;
+import com.google.instrumentation.trace.TraceConfig.TraceParams;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Random;
+import javax.annotation.Nullable;
+
+/** Implementation of the {@link SpanFactory}. */
+final class SpanFactoryImpl extends SpanFactory {
+  private static final EnumSet<Span.Options> noRecordEventOptions =
```

Comment:
I would remove both of the RecordEventOptions values. At the moment, we only have the one option, but in the future we may have more, in which case this code will need to be changed anyway - may as well assume that now. See comments later whn these values are used. to create spanOptions.